### ISSUE TITLE
Preparing mutations step is very slow

### ISSUE THREAD

**Issue by stanislaw** [mull-project/mull#583]
This is related to #582 but the focus here is on the `Preparing mutations` step which is also very slow.

```
Running original tests (threads: 8): 71/71. Finished in 3157ms.
Searching mutants across functions (threads: 8): 5923/5923. Finished in 69ms.
Applying filter: no debug info (threads: 8): 42737/42737. Finished in 13ms.
Applying filter: file path (threads: 8): 36099/36099. Finished in 384ms.
Applying filter: junk (threads: 8): 28140/28140. Finished in 93285ms.
Preparing mutations (threads: 1): 1/1. Finished in 227280ms. !!!
Applying mutations (threads: 1): 468/468. Finished in 33ms.
```

**Comment by AlexDenisov** [comment]
I split this step into several smaller ones and also parallelized them: #585 
That should improve the situation a bit. Another PR is coming later.

**Comment by AlexDenisov** [comment]
Fixed by #595
